__tests__/collection.test.ts

```diff
@@ -435,3 +435,28 @@ describe('random thisArg tests', () => {
 		}, array);
 	});
 });
+
+describe('ensure() tests', () => {
+	const coll = new Collection();
+	coll.set('a', 1);
+	coll.set('b', 2);
+
+	test('set new value if key does not exist', () => {
+		coll.ensure('c', () => 3);
+		expect(coll.size).toStrictEqual(3);
+		expect(coll.has('c')).toBeTruthy();
+	});
+
+	test('return existing value if key exists', () => {
+		const ensureB = coll.ensure('b', () => 3);
+		const getB = coll.get('b');
+		expect(ensureB).toBe(getB);
+		expect(coll.size).toStrictEqual(3);
+	});
+
+	test('ensure with existing key should not change the collection', () => {
+		coll.ensure('a', () => 4);
+		expect(coll.size).toStrictEqual(3);
```

Comment:
There are only two entries, since no new one is added:
```suggestion
		expect(coll.size).toStrictEqual(2);
```

Reply:
This requested change expectedly caused the test to fail.
![image](https://user-images.githubusercontent.com/22574513/143299803-8324bc41-db20-4551-9dfb-817fd48af7d7.png)
Line 445 expectedly adds another item to the collection, increasing its size to `3`
```js
coll.ensure('c', () => 3);
```

Reply:
Oh... then make those tests scope-based instead of sharing them, please.

Reply:
👍 applied request changes